src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,9 +8,16 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Address should be 4 strings seperated by commas, representing block" +
+            "(number and character only), street(number and character only), unit (number and character and #), and postal code" +
+            "(fixed 6 numbers)";
+    public static final String ADDRESS_VALIDATION_REGEX = "[0-9a-zA-Z]+,\\s[0-9a-zA-Z\\s]+,\\s[0-9#a-zA-Z-]+,\\s(\\d{6})";
+
+    public final Block block;
+    public final Street street;
```

Comment:
Please reconsider whether to use Public or Private classes here. You may refer this link for compositions https://www.journaldev.com/1325/composition-in-java-example